cmd/kyma/alpha/init/module/module.go

```diff
@@ -0,0 +1,72 @@
+package module
+
+import (
+	"fmt"
+
+	"github.com/kyma-project/cli/internal/cli"
+	"github.com/kyma-project/cli/pkg/module/scaffold"
+	"github.com/mandelsoft/vfs/pkg/osfs"
+	"github.com/spf13/cobra"
+)
+
+type command struct {
+	opts *Options
+	cli.Command
+}
+
+//NewCmd creates a new Kyma CLI command
+func NewCmd(o *Options) *cobra.Command {
+
+	c := command{
+		Command: cli.Command{Options: o.Options},
+		opts:    o,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "module MODULE_NAME <PARENT_DIR>",
```

Comment:
Do you think it's more usable both for the user and on the implementation side?
I'd say an explicit module name is better. Let's consider three examples:
- `kyma alpha init module foo` => module foo, assume current dir
- `kyma alpha init module /a/b/foo` => module foo, but it's implicit - user must parse the path with their eyes to see the module name - maybe the path is very long and copied from the clipboard?
- `kyma alpha init module a/b/foo` (no leading slash) => report error?

I really prefer current syntax more, if anyone wants to use current dir, there's a tool for that already:

`kyma alpha init module foo $PWD` 

Maybe we make `<PARENT_DIR>` optional and when it's not provided - we assume current directory? I was thinking about it, but I thought it will be a source of errors for unexperienced users.

Reply:
I don't see the advantage of the 2 args option. For me it is easier to read `/a/b/foo` than `foo /a/b`. Also having 2 args is much more error prone (e.g. users can mix the order and end up having a module called `/a/b` inside a `foo` folder) and needs more validation.

Regarding the `a/b/foo` case assume current dir too and create nested structure (`os.MkdirAll()` handles all 3 cases for you without problem).

And for using pwd, you can do `kyma alpha init module $PWD/foo` as far as I know.

Reply:
Well, in my opinion the explicit params are always better than combined or "implicit" ones, because the latter are usually confusing for new users. Experienced users will be fine with both syntaxes.
 
If the module name is explicit, then user have no doubts what is the module name - because it's given explicitly.
If the module name is embedded in the path `/a/b/foo`, then users may think that the entire thing is the module name: `/a/b/foo`. Hierarchical names are very common after all.

Maybe we should approach this issue by looking at the following problem: "How to explain to the user what is the tool  doing?"

In the "two params syntax", users must know the syntax, of course, but it's also the case with every CLI tool that exists.
And we only have to explain that: `kyma alpha init module foo /a/b` creates a directory for the module "foo" as a subdirectory of "/a/b". No other cases.

In the "combined syntax": `kyma alpha init module /a/b/foo`, we have to first tell the users that last path element is the module name - because it's important.
Then we have to document all the cases we discussed above.

That's why I think that it's actually easier to have two parameters, because it's easier to explain how it works. No different cases, no implicit behavior.

Having said that, I don't think it's really a big issue. As I said, it may only be a problem for new users. Experienced ones will learn the convention.

Reply:
After thinking about it for a while, I think the best solution would be to remove the path at all from the positional parameters.
- `kyma alpha init module foo` creates the foo directory in the current working dir.
- `kyma alpha init module foo --path=/a/b` creates the foo directory in the given location, that must exist.

No ambiguity, no way to mix the order of parameters, simple to explain/document.

Reply:
IMHO developers will not be confused by passing a module path and in general I try to keep things simple; the less number of arguments and the more the tool can automatically detect the better.

Reply:
Have not clear favourite for this approach. I would also go to use the plain module-name and use the current directory. If needed, the user can set the `--path` parameter if we want to use a different directory.

Reply:
I just had a look at the `init function` command and for consistency we could follow the same approach:
https://github.com/kyma-project/cli/blob/0972b8f3088b84bb083fa4a6112eb1b1107334ee/cmd/kyma/init/function/function.go#L50

Only difference is that the `--name` flag would be mandatory and the command would fail if not provided.

Additionally, we could validate that the module name has more than 3 chars and no special characters (except `-`, `_` or `.` ).

@Tomasz-Smelcerz-SAP What do you think about this approach? It would actually be very clse to your initial one but more explicit.

Reply:
OK, so the command would look like:
`kyma alpha init module --name=foo --path=/a/b` ? It's fine for me.